packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -4568,13 +4568,21 @@ abstract class ComponentElement extends Element {
   }
 }
 
+T unsafeCast<T>(Object any) => any; // ignore: return_of_invalid_type
+
 /// An [Element] that uses a [StatelessWidget] as its configuration.
 class StatelessElement extends ComponentElement {
   /// Creates an element that uses the given widget as its configuration.
   StatelessElement(StatelessWidget widget) : super(widget);
 
   @override
-  StatelessWidget get widget => super.widget as StatelessWidget;
+  StatelessWidget get widget {
+    if (kIsWeb) {
```

Comment:
The native compilers are not skipping the cast, regardless of whether you make it implicit or not, so why not just rename to `unsafeCastOnTheWeb` and eliminate the kIsWeb.  So long as `unsafeCastOnTheWeb` is inlined on native and dart2js (I think there are pragmas you can use to guarantee that) you should see:
  - No change in the native generated code
  - No change in dart2js spec mode code
  - The cast is unsafely ignored in dart2js production mode (-O4)